portability-core/src/test/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactsServiceTest.java

```diff
@@ -100,120 +66,105 @@ public void setup() {
+            .list(SELF_RESOURCE))
+            .thenReturn(listConnections);
+    when(listConnections.execute())
+            .thenReturn(listConnectionsResponse);
     when(peopleService.people()
-        .getBatchGet()
-        .setResourceNames(Collections.singletonList(RESOURCE_NAME))
-        .setPersonFields(PERSON_FIELDS)
-        .execute())
-        .thenReturn(batchResponse);
+            .getBatchGet()
+            .setResourceNames(Collections.singletonList(RESOURCE_NAME))
+            .setPersonFields(PERSON_FIELDS)
+            .execute())
+            .thenReturn(batchResponse);
 
     // Run test
     ContactsModelWrapper wrapper = contactsService.export(emptyExportInformation);
 
+    // Check the correct calls were made
+    verify(peopleService, atLeastOnce()).people().connections().list(SELF_RESOURCE);
+    verify(listConnections, atLeastOnce()).execute();
+    verify(peopleService, atLeastOnce()).people().getBatchGet()
+            .setResourceNames(Collections.singletonList(RESOURCE_NAME)).setPersonFields(PERSON_FIELDS)
+            .execute();
+
     // Check continuation information
     assertThat(wrapper.getContinuationInformation().getSubResources()).isEmpty();
     GooglePaginationInfo googlePaginationInfo = (GooglePaginationInfo) wrapper
-        .getContinuationInformation()
-        .getPaginationInformation();
+            .getContinuationInformation()
+            .getPaginationInformation();
     assertThat(googlePaginationInfo.getPageToken()).isEqualTo(nextPageToken);
 
     // Check that the right number of VCards was returned
     Collection<VCard> vCardCollection = wrapper.getVCards();
     assertThat(vCardCollection.size()).isEqualTo(connectionsList.size());
   }
 
-  private static <T extends VCardProperty, V> List<V> getValuesFromProperties(List<T> propertyList,
-      Function<T, V> function) {
-    return propertyList.stream().map(function).collect(Collectors.toList());
-  }
+  @Test
+  public void exportSubsequentPage() throws IOException {
+    List<Person> connectionsList = Collections.singletonList(PERSON);
+    String nextPageToken = "token";
+    ExportInformation nextPageExportInformation = new ExportInformation(Optional.empty(),
+            Optional.of(new GooglePaginationInfo(nextPageToken)));
+
+    ListConnectionsResponse listConnectionsResponse = new ListConnectionsResponse();
+    listConnectionsResponse.setConnections(connectionsList);
+    listConnectionsResponse.setNextPageToken(null);
 
-  private static <T extends TextProperty> List<String> getValuesFromTextProperties(
-      List<T> propertyList) {
-    return getValuesFromProperties(propertyList, T::getValue);
+    Connections.List listConnections = mock(Connections.List.class);
+
+    PersonResponse personResponse = new PersonResponse().setPerson(PERSON);
+    GetPeopleResponse batchResponse = new GetPeopleResponse().setResponses(Collections.singletonList(personResponse));
+
+    when(peopleService.people()
```

Comment:
Can we break this out into a separate method since it seems common between tests?

Reply:
I can break it up some.  How's this?